source/Octopus.Tentacle.Tests.Integration/MachineConfigurationHomeDirectoryTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Tentacle.CommonTestUtils;
+using Octopus.Tentacle.Tests.Integration.Support;
+using Octopus.Tentacle.Variables;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    [IntegrationTestTimeout]
+    public class MachineConfigurationHomeDirectoryTests : IntegrationTest
+    {
+        [Test]
+        [TentacleConfigurations(scriptServiceToTest: ScriptServiceVersionToTest.None)]
+        public async Task ShouldUseTheDefaultMachineConfigurationHomeDirectoryWhenACustomLocationIsNotProvided(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            await using (var clientAndTentacle = await tentacleConfigurationTestCase
+                             .CreateBuilder()
+                             .UseDefaultMachineConfigurationHomeDirectory()
+                             .Build(CancellationToken))
+            {
+                
+                var defaultMachineConfigurationHomeDirectory = PlatformDetection.IsRunningOnWindows ? 
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), "Octopus", "Tentacle", "Instances") : 
+                    "/etc/octopus/Tentacle/Instances";
+
+                var expectedInstanceConfigurationFilePath = new FileInfo(Path.Combine(defaultMachineConfigurationHomeDirectory, $"{clientAndTentacle.RunningTentacle.InstanceName}.config"));
+
+                expectedInstanceConfigurationFilePath.Exists.Should().BeTrue("Instance configuration file should exist");
```

Comment:
Nit: This will always be true on the build server due to this file existing from all the other test runs.

We could interrogate the file to see if it was updated? I.e. look for this instance in the file?

Reply:
It's looking for the instance config file for that running tentacle `${clientAndTentacle.RunningTentacle.InstanceName}.config"` am I missing something here ?

Reply:
🤦 No, you're right. I didn't see line 29... All good 😁